include/pacbio/gzfile/GZFileWriters.h

```diff
@@ -46,18 +46,38 @@
 #include "pbbam/BamRecord.h"
 #include "pbbam/DataSet.h"
 #include "pbbam/PbiRawData.h"
+#include "pacbio/gzfile/AbstractWriter.h"
 #include "pacbio/gzfile/GZFileWriter.h"
+#include "pacbio/gzfile/PlainFileWriter.h"
 
 namespace PacBio {
 namespace Postprimary {
 
-// wrapper around single or multiple GZFileWriters (when splitting by barcode)
-class GZFileWriters
+class AbstractWriterFactory {
```

Comment:
I chose that design pattern because you suggested it! Anyway, do you really want an explanation in the code? It seems self-explanatory.